scripts/linters/third_party_typings_linter.py

```diff
@@ -0,0 +1,151 @@
+    },
+    {
+        'name': 'Nerdamer',
+        'package.json': 'nerdamer',
+        'type_defs_file_name': 'nerdamer-defs-'
+    }
+]
+
+
+def check_third_party_libs_type_defs(verbose_mode_enabled):
+    """Checks the type definitions for third party libs
+    are up to date.
+    """
+    if verbose_mode_enabled:
+        python_utils.PRINT('Starting type defs check')
+        python_utils.PRINT('----------------------------------------')
+
+    with linter_utils.redirect_stdout(sys.stdout):
+        failed = False
+        summary_messages = []
+
+        with python_utils.open_file(_MANIFEST_JSON_FILE_PATH, 'r') as f:
+            manifest = json.load(f)['dependencies']['frontend']
+
+        with python_utils.open_file(_PACKAGE_JSON_FILE_PATH, 'r') as f:
+            package = json.load(f)['dependencies']
+
+        for third_party_lib in THIRD_PARTY_LIBS:
+            if 'manifest.json' in third_party_lib:
+                lib_version = (
+                    manifest[third_party_lib['manifest.json']]['version'])
+
+            if 'package.json' in third_party_lib:
+                lib_version = package[third_party_lib['package.json']]
+
+                if lib_version[0] == '^':
+                    lib_version = lib_version[1:]
+
+            files_in_typings_dir = os.listdir(
+                os.path.join(os.getcwd(), 'typings')
+            )
+
+            prefix_name = third_party_lib['type_defs_file_name']
+
+            files_with_prefix_name = []
+
+            for file_name in files_in_typings_dir:
+                if file_name.startswith(prefix_name):
+                    files_with_prefix_name.append(file_name)
+
+            if len(files_with_prefix_name) > 1:
+                python_utils.PRINT(
+                    'There are multiple type definitions for %s in the typings '
+                    'dir.' % third_party_lib['name']
+                )
+                python_utils.PRINT('')
+                failed = True
+            elif len(files_with_prefix_name) == 0:
+                python_utils.PRINT(
+                    'There are no type definitions for %s in the typings '
+                    'dir.' % third_party_lib['name']
+                )
+                python_utils.PRINT('')
+                failed = True
+            else:
+                type_defs_file_name = files_with_prefix_name[0]
+
+                type_defs_version = type_defs_file_name[len(prefix_name): -5]
+
+                if lib_version != type_defs_version:
+                    python_utils.PRINT(
+                        'Type definitions for %s are not up to date. The '
+                        'current version of %s is %s and the type definitions '
+                        'are for version %s.' % (
+                            third_party_lib['name'], third_party_lib['name'],
+                            lib_version, type_defs_version))
+                    python_utils.PRINT('')
+                    failed = True
+
```

Comment:
Assume I have added a new typedef file for any dependencies so how will we make sure that the typedef filename has version + it's included in this list? [Basically how this lint test will make sure this happens for new type def files?]

Reply:
We'll have to add the third party lib in the list.

Reply:
I think we won't be adding third party libs frequently. When we can simply add it to the list.

Reply:
But we should have this check for any new typings we add, right? (To have this test future proof)

Reply:
I am unable to think of a way to automate this check. I mean not all files in `package.json` o `manifest.json` have typedefs. So, how do we know which ones to check?